### PR TITLE
Pass _internalModel into a record when it is created

### LINKED ISSUE #3255: _internalModel is undefined on 'init' of a record.
Testcase: http://jsbin.com/cegapapigo/2/edit?html,css,js,output

```js
DS.Model.extend({
  foo: DS.attr('string'),
  initFoo: Ember.on('init', function() {
    this.get('foo');
  })
})._create();
```

### PR DESCRIPTION
**Pull request by bmac**
Closes #3255.

### CODE REVIEW

**Comment by igorT** [comment]
Any way we could have a test please?

**Comment by bmac** [comment]
Will do

**Comment by bmac** [comment]
I added a test. @igorT let me know if there is a better way to test this without directly asserting around internals.

File: packages/ember-data/tests/unit/model-test.js [hunk 1]
```diff
@@ -869,3 +869,23 @@ test('toJSON looks up the JSONSerializer using the store instead of using JSONSe
...
+
+
+test('_internalModel should be set by the store when a record is created', function() {
+  expect(1);
+  var Person = DS.Model.extend({
+    name: DS.attr(),
+
+    init: function() {
+      this._super.apply(this, arguments);
+      ok(this._internalModel);
```
Comment (igorT) [inline packages/ember-data/tests/unit/model-test.js]: ok(this.get('name')) is the thing we actually care about

Comment (bmac) [inline packages/ember-data/tests/unit/model-test.js]: updated

**Comment by igorT** [comment]
cc @fivetanley for beta.19.1

**Pull request merged**